Delete Feature - UI

amelendez37 opened the pull request.

amelendez37 committed:
Fixed issue with image file names and webpack

amelendez37 committed:
workout components scrollable

amelendez37 committed:
Added workout seed data

amelendez37 committed:
Commit for rebase

amelendez37 committed:
Added workout modal container

amelendez37 committed:
Committing for rebase

amelendez37 committed:
Completed workouts modal and disconnected workout list item from redux store

amelendez37 committed:
Added delete query for usersWorkouts table

jaba9531 reviewed (approved):
Looks Good

zgallagher08 commented on client/src/actions/index.js:
```diff
@@ -102,3 +103,20 @@ export const selectedWorkout = (workout) => {
+        Authorization: `${document.cookie}`
+      }
+    });
+    const updatedWorkouts = await getWorkoutsList(userId);
```
you should just delete the one from the array and send it through this action to update the redux store instead

zgallagher08 commented on client/src/components/workoutsView/workoutsListItem.jsx:
```diff
@@ -1,38 +1,48 @@
+          <img className="float-left delete-icon" src={require('../../../dist/images/trash.png')} 
+           onClick={
+             async (e) => {
+               e.persist();
```
what does e.persist do?

zgallagher08 commented on client/src/components/workoutsView/workoutsListItem.jsx:
```diff
@@ -1,38 +1,48 @@
+           onClick={
+             async (e) => {
+               e.persist();
+               await this.handleWorkoutClick.bind(this)();
```
use **e.target.getAttribute('data')** to get the workout id, and you can send that to the action to delete the workout.

to add the data attribute to the trash can, 
<img className="float-left delete-icon" **data={this.props.workout.id}** src={require('../../../dist/images/trash.png')} />

zgallagher08 commented on client/src/containers/workoutsListContainer.jsx:
```diff
@@ -24,21 +28,32 @@ class WorkoutsListContainer extends Component {
   }
   
+  toggleModal(e) {
+    e.stopPropagation();
```
what does e.stopPropagation() do?

zgallagher08 replied:
this turned yellow, not sure why, but nevermind.

zgallagher08 reviewed (changes requested):
Looks good, just a couple fixes

amelendez37 committed:
Refactored delete workout functionality

No longer making get request after deleting a workout. Instead just iterating over workouts within store.

zgallagher08 commented on client/src/components/workoutsView/workoutModal.jsx:
```diff
@@ -12,11 +12,8 @@ class WorkoutModal extends Component {
-                 className="float-left modal-btn pointer"
-                 onClick={(e) => this.props.handleYesClick(e, this.props.clickedWorkout.id, this.props.userId)}
-                >Yes</a>
+                <p>Delete *get name*?</p>
```
can you replace *get name* with the name?

zgallagher08 commented on client/src/components/workoutsView/workoutsListItem.jsx:
```diff
@@ -35,14 +35,7 @@ class WorkoutsListItem extends Component {
-               this.props.toggleModal(e);
-             }
-           }></img>
+          <img className="float-left delete-icon" src={require('../../../dist/images/trash.png')} data={this.props.workout.id} onClick={(e) => this.props.toggleModal(e)}></img>
```
this is good, just for future reference:
onClick={(e) => this.props.toggleModal(e)}
is equivalent to
onClick={this.props.toggleModal}

zgallagher08 reviewed (approved):
Good, I left 2 comments (the first should be resolved, but not urgent)

amelendez37 committed:
Added data-name attribute to delete buttons

data-name contains the name of a workout which gets passed down to the workout modal
